File: PRACTICAS DE PROG 2/Practica 4/Figuras Geometricas.py
import math

class FigurasGeometricas:
    def __init__(self, valor1, valor2=0):
        self.valor1 = valor1
        self.valor2 = valor2
    def __pow__(self, other): 
        return math.pi * (self.valor1 ** 2)

    def __mul__(self, other): 
        return self.valor1 * self.valor2

    def __truediv__(self, other): 
        return (self.valor1 * self.valor2) / 2

    def __add__(self, other):  
        return ((self.valor1 + other.valor1) / 2) * self.valor2

    def __mod__(self, other):  
        apotema = self.valor2
        perimetro = 5 * self.valor1
        return (perimetro * apotema) / 2

circulo = FigurasGeometricas(7)
print("Área del círculo:", circulo ** None)

rectangulo = FigurasGeometricas(10, 5)
print("Área del rectángulo:", rectangulo * None)

triangulo = FigurasGeometricas(10, 5)
print("Área del triángulo:", triangulo / None)

trapecio = FigurasGeometricas(8, 5)
print("Área del trapecio:", trapecio + FigurasGeometricas(6, 5))

pentagono = FigurasGeometricas(5, 4) 
print("Área del pentágono:", pentagono % None)

File: PRACTICAS DE PROG 2/Practica 4/FigurasGeometricas.java
public class FigurasGeometricas {
    private double valor1;
    private double valor2;

    public FigurasGeometricas(double valor1, double valor2) {
        this.valor1 = valor1;
        this.valor2 = valor2;
    }

    public double areaCirculo() { 
        return Math.PI * Math.pow(valor1, 2);
    }

    public double areaRectangulo() { 
        return valor1 * valor2;
    }

    public double areaTriangulo() { 
        return (valor1 * valor2) / 2;
    }

    public double areaTrapecio(double base2) { 
        return ((valor1 + base2) / 2) * valor2;
    }

    public double areaPentagono() { 
        double perimetro = 5 * valor1;
        double apotema = valor2; 
        return (perimetro * apotema) / 2;
    }

    public static void main(String[] args) {
        FigurasGeometricas circulo = new FigurasGeometricas(7, 0);
        System.out.println("Área del círculo: " + circulo.areaCirculo());

        FigurasGeometricas rectangulo = new FigurasGeometricas(10, 5);
        System.out.println("Área del rectángulo: " + rectangulo.areaRectangulo());

        FigurasGeometricas triangulo = new FigurasGeometricas(10, 5);
        System.out.println("Área del triángulo: " + triangulo.areaTriangulo());

        FigurasGeometricas trapecio = new FigurasGeometricas(8, 5);
        System.out.println("Área del trapecio: " + trapecio.areaTrapecio(6));

        FigurasGeometricas pentagono = new FigurasGeometricas(5, 4);
        System.out.println("Área del pentágono: " + pentagono.areaPentagono());
    }
}